File: backend/app/models/fuel_record.py
"""燃費記録モデル."""

from datetime import datetime
from uuid import UUID

from sqlalchemy import Boolean, DateTime, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, SoftDeleteMixin, TimestampMixin, UUIDPKMixin


class FuelRecord(UUIDPKMixin, TimestampMixin, SoftDeleteMixin, Base):
    """燃費記録モデル."""

    __tablename__ = "fuel_record"

    vehicle_id: Mapped[UUID] = mapped_column(index=True)
    user_id: Mapped[UUID] = mapped_column(index=True)
    refuel_datetime: Mapped[datetime] = mapped_column(DateTime(timezone=True))
    total_mileage: Mapped[int] = mapped_column(Integer)
    fuel_type: Mapped[str] = mapped_column(String(50))
    unit_price: Mapped[int] = mapped_column(Integer)
    total_cost: Mapped[int] = mapped_column(Integer)
    is_full_tank: Mapped[bool] = mapped_column(Boolean, default=False)
    gas_station_name: Mapped[str | None] = mapped_column(String(255))

File: frontend/src/lib/types/fuel-record.ts
import type { SuccessResponse } from './api'

export interface FuelRecord {
  id: string
  vehicle_id: string
  user_id: string
  refuel_datetime: string
  total_mileage: number
  fuel_type: string
  unit_price: number
  total_cost: number
  is_full_tank: boolean
  gas_station_name: string | null
  distance_traveled: number | null
  fuel_amount: number | null
  fuel_efficiency: number | null
  created_at: string
  updated_at: string
  deleted_at?: string | null
}

export interface CreateFuelRecordInput {
  vehicle_id: string
  refuel_datetime: string
  total_mileage: number
  fuel_type: string
  unit_price: number
  total_cost: number
  is_full_tank?: boolean
  gas_station_name?: string | null
}

export interface UpdateFuelRecordInput {
  refuel_datetime?: string | null
  total_mileage?: number | null
  fuel_type?: string | null
  unit_price?: number | null
  total_cost?: number | null
  is_full_tank?: boolean | null
  gas_station_name?: string | null
}

export type FuelRecordResponse = SuccessResponse<FuelRecord>
export type FuelRecordsResponse = SuccessResponse<FuelRecord[]>
